File: backend/spring-boot/src/main/java/org/bugzkit/api/shared/config/MailConfig.java
package org.bugzkit.api.shared.config;

import java.util.Objects;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.mail.javamail.JavaMailSender;
import org.springframework.mail.javamail.JavaMailSenderImpl;

@Configuration
public class MailConfig {
  @Value("${spring.mail.host}")
  private String host;

  @Value("${spring.mail.port}")
  private String port;

  @Value("${spring.mail.username}")
  private String username;

  @Value("${spring.mail.password}")
  private String password;

  @Bean
  public JavaMailSender getJavaMailSender() {
    final var mailSender = createMailSender();
    setProperties(mailSender);
    return mailSender;
  }

  private JavaMailSenderImpl createMailSender() {
    final var mailSender = new JavaMailSenderImpl();
    mailSender.setHost(host);
    mailSender.setPort(Integer.parseInt(Objects.requireNonNull(port)));
    mailSender.setUsername(username);
    mailSender.setPassword(password);
    return mailSender;
  }

  private void setProperties(JavaMailSenderImpl mailSender) {
    final var props = mailSender.getJavaMailProperties();
    props.put("mail.transport.protocol", "smtp");
    props.put("mail.smtp.auth", "true");
    props.put("mail.smtp.starttls.enable", "true");
    props.put("mail.debug", "true");
  }
}

File: backend/spring-boot-kotlin/src/main/kotlin/org/bootstrapbugz/api/shared/config/MailConfig.kt
package org.bootstrapbugz.api.shared.config

import java.util.*
import org.springframework.beans.factory.annotation.Value
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.mail.javamail.JavaMailSender
import org.springframework.mail.javamail.JavaMailSenderImpl

@Configuration
class MailConfig(
  @Value("\${spring.mail.host}") private val host: String,
  @Value("\${spring.mail.port}") private val port: String,
  @Value("\${spring.mail.username}") private val username: String,
  @Value("\${spring.mail.password}") private val password: String
) {
  @Bean
  fun getJavaMailSender(): JavaMailSender {
    val mailSender = createMailSender()
    setProperties(mailSender)
    return mailSender
  }

  private fun createMailSender(): JavaMailSenderImpl {
    return JavaMailSenderImpl().also { sender ->
      sender.host = this.host
      sender.port = this.port.toInt()
      sender.username = this.username
      sender.password = this.password
    }
  }

  private fun setProperties(mailSender: JavaMailSenderImpl) {
    mailSender.javaMailProperties.apply {
      put("mail.transport.protocol", "smtp")
      put("mail.smtp.auth", "true")
      put("mail.smtp.starttls.enable", "true")
      put("mail.debug", "true")
    }
  }
}
